rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,426 @@
+
+## CommonMark
+
+**CommonMark is the content of all doc-comments.**
+
+Adopting CommonMark as the content for all doc-comments brings the benefits of widely accepted and understood documentation format in tech projects, while maintaining profitability and consistency within the Nix ecosystem by aligning with existing [NixOS/RFC-72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).
+
+## `/** */` is the doc-comment format
+
+The decision to use /** to start a doc-comment ensures a unique distinction from regular comments while still allowing seamless writing without IDE or editor support. A single choice provides the best developer experience and simplifies tooling.
+
+## Placement
+
+**The placement describes the relationship between doc-comments and the expression they are documenting.**
+
+- Doc-comments are placed before the documentable node. Only whitespace is allowed in between. ([Examples](#basic-examples)) 
+
+- The documentation present before the `attribute path` describes the body of the attribute. ([Examples](#Attributes))
+    - In case placement is ambiguous, the one closer to the body has higher precedence. ([Examples](#ambiguous-placement))
+         
+- All partial functions of a curried lambda can share the same placement with the outermost lambda. ([Examples](#partial-lambda-functions))
+  
+> Note: Research of the RFC Sheperds Team showed that this allows for intuitive placements like are already done in nixpkgs.
+
+### Examples
+
+#### Basic examples
+
+Only whitespaces between the `documentable node` and the `doc-comment`
+
+```nix
+/**Doc for anonymous lambda function*/
+↓
+x: x;
+```
+
+```nix
+listToAttrs [
+  { name = "foo"; value = /**Documentation for '1'*/1; }
+]
+```
+
+#### Attributes 
+
+It is allowed to write the documentation before the attribute instead of placing it right before the body.
+
+```nix
+/**Doc for lambda function bound to a variable*/
+           ↓
+assigned = x: x;
+```
+
+```nix
+{
+    /**This documents the specialisation `map (x: x)` */
+          ↓
+    foo = map (x: x);          
+}
+```
+
+#### Ambiguous placement
+
+```nix
+/**Doc B*/
+int = /**Doc A*/1;
+
+# Documentation is 'Doc A' because it is directly next to the documentable body.
+```
+
+#### Dynamic attribute
+
+```nix
+{
+    /** Documentation for '2' */
+                                   ↓
+    ${let name = "bar"; in name} = 2;
+}
+# Dynamic attribute
+```
+
+#### Partial lambda functions
+
+```nix
+/**Generic doc for all partial lambda functions*/
+↓1 ↓2          ↓3
+x: ({y, ...}:  z: x + y) * z;
+
+# All partial functions can share the same placement with the outermost lambda.
+# 1 -> Lambda Value `x: ({y, ...}: z: x + y) * z` 
+# 2 -> Lambda Value `({y, ...}: z: x + y) * z`    
+# 3 -> Lambda Value `z: x + y) * z`               
```

Comment:
This seems odd to me, why is this necessary? Wouldn't it make more sense to just attach the doc comment to only the first lambda? Or alternatively, why does this only work for lambda's but not other expressions?

PS: There's a misplaced `)` here, should be at the end.

Reply:
Yes overall you would usually write some documentation for the specialized lambda. We thought of adding it as a fallback for spezialized lamdas when documentation is missing or cannot be written elsewhere. 
You still see that the documented function is already applied e.g., 2 times but there is not other source of documentation. In this case you can decide to display this "fallback documentation" of the first lambda. Its still an edge case and shouldnt be overused because it is less correct.